From: Frank Reininghaus <frank78ac@googlemail.com>
Date: Mon, 23 Jan 2012 18:28:21 +0000
Subject: Make sure that Control+click toggles the selection state
X-Git-Url: http://quickgit.kde.org/?p=kde-baseapps.git&amp;a=commitdiff&amp;h=e8bfc8724b441b70e440cad05983134975facc8b
---
Make sure that Control+click toggles the selection state

This commit fixes a regression caused by the recent commit
9f711b5f2e1d1fd856cd6b033e6adb96f9b46d8a.
CCBUG: 292250
(cherry picked from commit 84a9cc4bf6e9decc4c102102c4b04162369eb0fe)
---


--- a/dolphin/src/kitemviews/kitemlistcontroller.cpp
+++ b/dolphin/src/kitemviews/kitemlistcontroller.cpp
@@ -471,7 +471,7 @@ bool KItemListController::mousePressEven
                                 (!shiftOrControlPressed && !pressedItemAlreadySelected);
     if (clearSelection) {
         m_selectionManager->clearSelection();
-    } else if (pressedItemAlreadySelected && (event->buttons() & Qt::LeftButton)) {
+    } else if (pressedItemAlreadySelected && !shiftOrControlPressed && (event->buttons() & Qt::LeftButton)) {
         // The user might want to start dragging multiple items, but if he clicks the item
         // in order to trigger it instead, the other selected items must be deselected.
         // However, we do not know yet what the user is going to do.
